modules/base/src/processors/imagestackvolumesource.cpp

```diff
@@ -0,0 +1,372 @@
+        }
+    }
+    if (!fileNames.size()) {
+        LogWarn("No images found in '" << filePattern_.getFilePatternPath() << "'");
+        return;
+    }
+
+    using ReaderMap = std::map<std::string, std::unique_ptr<DataReaderType<Layer>>>;
+    ReaderMap readerMap;
+
+    size_t numSlices = 0;
+    // determine number of slices and file readers for all images
+    for (auto file : fileNames) {
+        std::string fileExtension = toLower(filesystem::getFileExtension(file));
+        if (readerMap.find(fileExtension) != readerMap.end()) {
+            // reader already exists for this format
+            ++numSlices;
+            continue;
+        }
+        auto reader = InviwoApplication::getPtr()
+                          ->getDataReaderFactory()
+                          ->getReaderForTypeAndExtension<Layer>(fileExtension);
+        if (!reader && skipUnsupportedFiles_.get()) {
+            // ignore file entirely
+            continue;
+        }
+        readerMap[fileExtension] = std::move(reader);
+
+        ++numSlices;
+    }
+
+    if (readerMap.empty()) {
+        // could not find any suitable data reader for the images
+        LogWarn("Image formats not supported");
+        return;
+    }
+
+    auto getReaderIt = [&readerMap](const std::string& filename) {
+        return readerMap.find(toLower(filesystem::getFileExtension(filename)));
+    };
+
+    // use first image to determine the volume size
+    size2_t layerDims{0u, 0u};
+    //    const DataFormatBase* format = nullptr;
+    try {
+        size_t slice = 0;
+
+        // skip slices with unsupported image formats
+        while (getReaderIt(fileNames[slice]) == readerMap.end()) {
+            ++slice;
+        }
+
+        auto imgLayer = getReaderIt(fileNames[slice])->second->readData(fileNames[slice]);
+        // Call getRepresentation here to force read a ram representation.
+        // Otherwise the default image size, i.e. 256x265, will be reported
+        // until you do the conversion since the LayerDisk does not provide any metadata.
+        auto layerRAM = imgLayer->getRepresentation<LayerRAM>();
+        layerDims = imgLayer->getDimensions();
+
+        if ((layerDims.x == 0) || (layerDims.y == 0)) {
+            LogError("Could not extract valid image dimensions from '" << fileNames[slice] << "'");
+            return;
+        }
+
+        volume_ = layerRAM->dispatch<std::shared_ptr<Volume>, FloatOrIntMax32>(
+            [&](auto baselayerprecision) {
+                using ValueType = util::PrecsionValueType<decltype(baselayerprecision)>;
+                using PrimitiveType = typename DataFormat<ValueType>::primitive;
+
+                const size3_t volSize(layerDims, numSlices);
+                // create matching volume representation
+                auto volumeRAM = std::make_shared<VolumeRAMPrecision<ValueType>>(volSize);
+
+                auto volume = std::make_shared<Volume>(volumeRAM);
+                volume->dataMap_.dataRange =
```

Comment:
Future feature: option to automatically decide ranges based on min/max. Especially for other formats, i.e. float